auth0/nextjs-auth0
# Bail out of static rendering for pages and routes in app dir

### 📋 Changes


Ensure `getConfig` doesn't run when the Next.js app is being statically rendered (because static rendering happens at build time and users don't necessarily want to supply the config at build time)

The 2 large commits are for making sure the config can be lazily evaluated at request time (`config` -> `getConfig(req)`):

- 30c100e3fdaa00be4d627f84351e3421d9829243 allow lazy config in auth0-session 
- d79ff8f59accdf0535d616ae80dc57de343317e9 allow lazy config in Next.js layer

The 3rd small commit is using this to bail out of static rendering before getting the config.

bec7d6c5998ba06ae87cb906235d7568bab2418f - Bail out of static rendering for pages and routes in app dir

You can bail out of static rendering by using the `cookies()` helper for Server Components and inspecting the request's `url` for API routes

### 📎 References

fixes #1356 
https://github.com/vercel/next.js/issues/49006

### 🎯 Testing

- Make sure you have no environment variables
- Run "npm run build:example"
- You should see no errors
